forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -4515,6 +4515,17 @@ public final Trigger getTriggerForStaticAbility(final String str, final StaticAb
         return result;
     }
 
+    public final Trigger addTriggerForStaticAbility(final Trigger trig, final StaticAbility stAb) {
+        String str = trig.toString();
+        Trigger result = storedTrigger.get(stAb, str);
+        if (result == null) {
+            result = trig.copy(this, false);
+            addTrigger(result);
```

Comment:
you shouldn't need to add the trigger here manually, that the job for the static